lec06-pop-models.Rmd

```diff
@@ -2,3 +2,298 @@
+
+The process we just used to get this model is probably not what you'd do in practice - 
+I just wanted to illustrate the connection between exponential growth and this differential equation.
+
+### Individual exercise
+
+If $r$ is made larger, does the population grow faster or slower? 
+
+## Analyzing models (qualitatively at first)
+
+When you analyze a model, the exact questions you answer will depend on the model. 
+But in general, the following questions are ones you will try to answer for any model.
+
+1. What are the **fixed points** of the model? (For which sizes of the population(s) does the population size remain constant in time?) 
+2. Which of the fixed points are **stable**? (If the population is slightly different from the fixed point, will it go towards it or away from it?)
+3. How do the above depend on parameters of the model?
+
+---------
+
+We will see how to answer these questions quantitatively soon. But let's start by looking at them
+qualitatively. We will draw a **phase portrait** of this system, and the techniques we use will be applicable
+to almost any model you might come across. 
+
+Let's start answering these questions qualitatively for the exponential growth model. There's a lot of
+information in the differential equation we can qualitatively read off. If we plot $dN/dt$ vs $N$, 
+the places where $dN/dt = 0$ are the fixed points of our system. Let's do this for the exponential growth model. 
+
+```{r}
+N_seq <- seq(-20,20) # make a sequence of N values to plot dN/dt against
+dN_dt = N_seq  # assume r = 1
+
+plot(N_seq,dN_dt)
+lines(N_seq,0*N_seq) # a line at zero for visual aid
+```
+We can also tell whether the fixed points are stable or unstable by looking at the sign of $dN/dt$ 
+on either side of the fixed point. For a one-dimensional system, our phase portrait is a line of $N$. 
+We mark the fixed point(s) with circles, and we draw an arrow to the left wherever $dN/dt$ is negative, 
+and an arrow to the right wherever $dN/dt$ is positive in the regions between fixed points. 
+
+
+<!-- diagram of phase portrait --> 
+
+
+This model is a bit boring - there's only one fixed point, and it's unstable. This makes sense - 
+we are talking about exponential growth, after all. There's no upper limit. 
+
+--------
+
+Of course in real life there must be an upper limit to growth, otherwise the surface of the earth would be
+covered in bacteria several feet thick in a matter of hours. Let's do a quick calculation for fun - how long
+would it take for a single bacterial cell undergoing exponential growth to take over the world?
+
+```{r}
+cell_volume = 0.6 #micrometres cubed - E. coli cell volume is ~0.6 to 0.7 micrometres^3
+growth_rate = 3 # let's say cells divide three times per hour - every 20 minutes (max growth rate of E. coli in good food conditions)
+
+N_t <- function(N0,r,t) { # r is growth rate - divisions per hour, t is time in hours, N0 is starting population
```

Comment:
Generally, if we are going to be consistent with out own style guide, a space should come after the `,`. You can also highlight the code and type `Ctrl-Shift-A` and the code will automatically be (re-)formatted.